builder/format.go

```diff
@@ -0,0 +1,130 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/pkg/errors"
+)
+
+// UpdateFormatVersion updates the builder.conf file with a new format version
+func UpdateFormatVersion(b *Builder, version string) error {
+	b.Config.Swupd.Format = version
+
+	newver := "${1}" + b.Config.Swupd.Format
+	var re = regexp.MustCompile(`(FORMAT=)[0-9]*`)
+
+	builderData, err := ioutil.ReadFile(b.BuildConf)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read builder.conf")
+	}
+
+	builderEdit := re.ReplaceAllString(string(builderData), newver)
+
+	var filename string
+	if UseNewConfig {
+		filename, err = GetConfigPath("")
+		if err != nil {
+			return err
+		}
+
+		return b.Config.SaveConfig(filename)
+	}
+
+	builderOut := []byte(builderEdit)
+	if err = ioutil.WriteFile(b.BuildConf, builderOut, 0644); err != nil {
+		return errors.Wrap(err, "Failed to write new builder.conf")
+	}
+
+	return nil
+}
+
+// CopyFullGroupsINI copies the initial ini file which has ALL bundle definitions
+func CopyFullGroupsINI(b *Builder) error {
+	return helpers.CopyFile(filepath.Join(b.Config.Builder.ServerStateDir, "full_groups.ini"), filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini"))
+}
+
+// CopyTrimmedGroupsINI copies the new ini made with deleted bundles removed
+func CopyTrimmedGroupsINI(b *Builder) error {
+	return helpers.CopyFile(filepath.Join(b.Config.Builder.ServerStateDir, "trimmed_groups.ini"), filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini"))
+}
+
+// RevertFullGroupsINI copies back the full ini to the manifest creator accounts for deleted bundles
+func RevertFullGroupsINI(b *Builder) error {
+	return helpers.CopyFile(filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini"), filepath.Join(b.Config.Builder.ServerStateDir, "full_groups.ini"))
+}
+
+// RevertTrimmedGroupsINI copies back the trimmed INI so manifests are not created anymore for deleted bundles in new format
+func RevertTrimmedGroupsINI(b *Builder) error {
+	return helpers.CopyFile(filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini"), filepath.Join(b.Config.Builder.ServerStateDir, "trimmed_groups.ini"))
+}
+
+func getLastBuildVersion(b *Builder) (string, error) {
+	var lastVer []byte
+	var err error
+
+	filename := filepath.Join(b.Config.Builder.ServerStateDir, "image/LAST_VER")
+	// Likely the first build
+	if lastVer, err = ioutil.ReadFile(filename); os.IsNotExist(err) {
+		return "", errors.Wrap(err, "Cannot find last built version, must get container image")
```

Comment:
Wait, we need the opposite, right? `os.IsNotExist(err)` will tell us that it _is_ a "file does not exist" error. So we need something like

```golang
func getLastBuildVersion(b *Builder) (string, error) {
	filename := filepath.Join(b.Config.Builder.ServerStateDir, "image/LAST_VER")
	// Likely the first build
	lastVer, err := ioutil.ReadFile(filename)
	if os.IsNotExist(err) {
		return "", nil
	} else if err != nil {
		return "", errors.Wrap(err, "Cannot find last built version, must get container image")
	}
	return strings.Split(string(lastVer), "\n")[0], nil
}
```

Either way, I still don't understand the container part of the error message -- I'm maybe missing something, 'cause this seems tangential to whether or not it's run natively.